contract/src/common.rs

```diff
@@ -124,6 +124,33 @@ pub(crate) mod tests {
             }
         }
 
+        pub(crate) fn with_products(mut self, products: &[Product]) -> Self {
+            for product in products {
+                self.contract.products.insert(product.clone().id, product.clone());
+            }
```

Comment:
There is no need to clone the whole product and then take only `id`. Just clone the id.
```suggestion
                self.contract.products.insert(product.id.clone(), product.clone());
            }
```